privacyidea/lib/tokens/emailtoken.py

```diff
@@ -328,6 +328,7 @@ def create_challenge(self, transactionid=None, options=None):
 
     @log_with(log)
     @check_token_locked
+    @check_token_otp_lenght
```

Comment:
please move this decorator one line up. Decorators are applied from bottom to top (innermost first). We should first check if the token is locked and then if the otp length matches.